src/lib/components/Cytoscape.react.js

```diff
@@ -297,6 +302,126 @@ class Cytoscape extends Component {
         });
 
+        this.createMenuItems = (ctxMenu) => {
+            const updateContextMenuData = (newContext) => {
+                this.props.setProps({contextMenuData: newContext});
+            };
+            const contextMenuDefaultFunctions = {
+                remove: function (event) {
+                    let target = event.target || event.cyTarget;
+                    target.remove();
+                },
+                add_node: function (event) {
+                    let pos = event.position || event.cyPosition;
+                    cy.add({
+                        data: {
+                            group: 'nodes',
+                        },
+                        position: {
+                            x: pos.x,
+                            y: pos.y,
+                        },
+                    });
+                },
+                add_edge: function (event) {
+                    const selectedNodeIds = selectedNodes.map((node) =>
+                        node.id()
+                    );
+                    if (selectedNodes.length === 0) {
+                        console.error(
+                            'Error: No nodes selected, cannot add edge'
+                        );
+                    } else if (selectedNodes.length === 1) {
+                        cy.add({
+                            data: {
+                                id: uuidv4(),
+                                group: 'edges',
+                                source: selectedNodeIds[0],
+                                target: selectedNodeIds[0],
+                            },
+                        });
+                    } else if (selectedNodes.length === 2) {
+                        cy.add({
+                            data: {
+                                id: uuidv4(),
+                                group: 'edges',
+                                source: selectedNodeIds[0],
+                                target: selectedNodeIds[1],
+                            },
+                        });
+                    } else {
+                        console.error(
+                            'Error: more than 2 nodes selected, cannot add edge'
+                        );
+                    }
+                },
+            };
+            const newMenuItems = [];
+            for (const item of ctxMenu) {
+                let onClickFunction;
+                let new_item = {};
+                // use default javascript function
+                if (contextMenuDefaultFunctions.hasOwnProperty(item.onClick)) {
+                    onClickFunction = contextMenuDefaultFunctions[item.onClick];
+                }
+                // use user defined javascript function in a namespace under assets/
+                else if (
+                    window.hasOwnProperty('dashCytoscapeFunctions') &&
+                    window.dashCytoscapeFunctions.hasOwnProperty(
+                        item.onClickCustom
```

Comment:
Same q -- I think this will cause an error if `item` doesn't have an `onClickCustom` property